pkg/web/apps/apps_psql_repository_test.go

```diff
@@ -1,56 +1,234 @@
+	getAppVersionsQueryString = `SELECT id,version,app_id,disabled,disabled_message,num_of_clients,num_of_app_launches FROM version WHERE app_id = \$1`
+)
+
+func Test_appsPostgreSQLRepository_GetApps_WillReturnTwoApps(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("Unexpected error opening a stub database connection: %v", err)
+	}
+
+	defer db.Close()
+
+	mockApps := []models.App{
+		models.App{
+			ID:      "7f89ce49-a736-459e-9110-e52d049fc025",
+			AppID:   "com.aerogear.mobile_app_one",
+			AppName: "Mobile App One",
+		},
+		models.App{
+			ID:      "7f89ce49-a736-459e-9110-e52d049fc026",
+			AppID:   "com.aerogear.mobile_app_three",
+			AppName: "Mobile App Two",
+		},
+		models.App{
+			ID:      "7f89ce49-a736-459e-9110-e52d049fc027",
+			AppID:   "com.aerogear.mobile_app_three",
+			AppName: "Mobile App Three",
+		},
+	}
+
+	cols := []string{"id", "app_id", "app_name", "deleted_at"}
+
+	timestamp := "2019-02-15T09:38:33+00:00"
+
+	// Insert an app where the deleted_at column is set
+	sqlmock.NewRows(cols).AddRow(mockApps[0].ID, mockApps[0].AppID, mockApps[0].AppName, timestamp)
+
+	// Insert 2 apps which are not soft deleted
+	rows := sqlmock.NewRows(cols).AddRow(mockApps[1].ID, mockApps[1].AppID, mockApps[1].AppName, "").AddRow(mockApps[2].ID, mockApps[2].AppID, mockApps[2].AppName, "")
+
+	// We should expected to get back only the apps which are not soft deleted
+	mock.ExpectQuery(getAppsQueryString).WillReturnRows(rows)
+	a := NewPostgreSQLRepository(db)
+
+	apps, err := a.GetApps()
+
+	if err != nil {
+		t.Fatalf("Got error trying to get apps from database: %v", err)
+	}
+
+	if len(*apps) != 2 {
+		t.Fatalf("Expected 2 apps to be returned from the database, got %v", len(*apps))
+	}
+}
 
+func Test_appsPostgreSQLRepository_GetApps_WillReturnNoApps(t *testing.T) {
+	db, mock, err := sqlmock.New()
 	if err != nil {
-		t.Errorf("Connect() returned an error: %v", err.Error())
-	}
-
-	tests := []struct {
-		name    string
-		want    *[]models.App
-		wantErr bool
-	}{
-		{
-			name: "appsPostgreSQLRepository.GetApps() should return a list of apps",
-			want: &[]models.App{
-				models.App{
-					ID:                    "a0874c82-2b7f-11e9-b210-d663bd873d93",
-					AppID:                 "com.aerogear.app1",
-					AppName:               "app1",
-					NumOfDeployedVersions: 1,
-					NumOfAppLaunches:      1,
-					NumOfClients:          1,
-				},
-			},
-			wantErr: false,
-		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			a := NewPostgreSQLRepository(dbConn)
-			got, err := a.GetApps()
-			if (err != nil) != tt.wantErr {
-				t.Errorf("appsPostgreSQLRepository.GetApps() error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("appsPostgreSQLRepository.GetApps() = %v, want %v", got, tt.want)
-			}
-		})
+		t.Fatalf("Unexpected error opening a stub database connection: %v", err)
+	}
+
+	defer db.Close()
+
+	mockApps := []models.App{
```

Comment:
Is not the same mock twice in the code? If yes, wdyt over centralize it? Wdyt over add a global var with this value? in fact, shows that it could be in fixtures.go. Wdyt?

Reply:
@camilamacedo86 yes this makes sense. I will do this now.